ospdo db migration docs (#715)
openstack-k8s-operators/data-plane-adoption: docs_user/modules/proc_retrieving-topology-specific-service-configuration.adoc

* To compare your {OpenStackShort} database with the adopted {rhos_acro} database

.Prerequisites
// Set OSPdO different values to be used in same commands
// When there is an OSPdO source env, some commands run on RHOSO namespace

> // and some on the OSPdO namespace with some added, different parameteres

Reviewer: Why is this text hidden?

. Export the shell variables for the following outputs and test the connection to the {OpenStackShort} database:
+
----

> export PULL_OPENSTACK_CONFIGURATION_DATABASES=$(oc run mariadb-client {pod_annotations} -q --image ${MARIADB_IMAGE} -i --rm --restart=Never -- \

Reviewer: `{pod_annotations}` applies to all deployments, not just OSPdO?

    mysql -rsh "$SOURCE_MARIADB_IP" -uroot -p"$SOURCE_DB_ROOT_PASSWORD" -e 'SHOW databases;')
echo "$PULL_OPENSTACK_CONFIGURATION_DATABASES"
----
endif::[]
ifeval::["{OpenStackPreviousInstaller}" == "director_operator"]
----
CONTROLLER1_SSH="oc -n $OSPDO_NAMESPACE rsh -c openstackclient openstackclient ssh controller-0.ctlplane"
export CONTROLLER1_SSH
----

> * With OSPdO, the mariadb-client needs to run on the same OCP node where the OSP Controller is running.  In addition, the internalapi-static network needs to be attached to the pod.

Reviewer: ```suggestion
* With OSPdO, the `mariadb-client` needs to run on the same {rhocp_long} node where the {OpenStackShort} Controller node is running.  In addition, the `internalapi-static` network needs to be attached to the pod.
```

----
PASSWORD_FILE="tripleo-passwords.yaml"
export PASSWORD_FILE
OSPDO_NAMESPACE="openstack"
export OSPDO_NAMESPACE
----

> . Get the passwords file
> ----
> oc get secret tripleo-passwords -n $OSPDO_NAMESPACE -o json | jq -r '.data["tripleo-overcloud-passwords.yaml"]' |
> base64 -d >"${PASSWORD_FILE}"

Reviewer: ```suggestion
. Get the passwords file:
+
----
$ oc get secret tripleo-passwords -n $OSPDO_NAMESPACE -o json | jq -r '.data["tripleo-overcloud-passwords.yaml"]' |
base64 -d >"${PASSWORD_FILE}"
```

> ----
> . Get the name of the OCP node where the OSP Controller VM is running
> ----
> CONTROLLER_NODE=$(oc get vmi -ojson | jq -r '.items[0].status.nodeName')
> export CONTROLLER_NODE
> SOURCE_OVN_OVSDB_IP=172.17.0.160 # TODO - get this from the source OVN DB
> export SOURCE_OVN_OVSDB_IP
> 
> SOURCE_DB_ROOT_PASSWORD=$(grep <"${PASSWORD_FILE}" ' MysqlRootPassword:' | awk -F ': ' '{ print $2; }') || {
>     echo "Failed to get the source DB root password"
>     exit 1
> }
> export SOURCE_DB_ROOT_PASSWORD
> ----

Reviewer: ```suggestion
----

. Get the name of the {OpenShiftShort} node where the {OpenStackShort} Controller virtual machine is running:
+
----
$ CONTROLLER_NODE=$(oc get vmi -ojson | jq -r '.items[0].status.nodeName')
export CONTROLLER_NODE
SOURCE_OVN_OVSDB_IP=172.17.0.160 # TODO - get this from the source OVN DB
export SOURCE_OVN_OVSDB_IP

SOURCE_DB_ROOT_PASSWORD=$(grep <"${PASSWORD_FILE}" ' MysqlRootPassword:' | awk -F ': ' '{ print $2; }') || {
    echo "Failed to get the source DB root password"
    exit 1
}
export SOURCE_DB_ROOT_PASSWORD
----
```

Author: ok

> . In OSPdO, the mysql service iP can be found in the tripleo-exports-default ConfigMap, section ctlplane-export.yaml
> ----
> cpexport=$(oc -n "${OSPDO_NAMESPACE}" get cm tripleo-exports-default -o json | jq -r '.data["ctlplane-export.yaml"]')

Reviewer: ```suggestion
. Find the mysql service IP in the `ctlplane-export.yaml` section of the `tripleo-exports-default` ConfigMap:
+
----
$ cpexport=$(oc -n "${OSPDO_NAMESPACE}" get cm tripleo-exports-default -o json | jq -r '.data["ctlplane-export.yaml"]')
```

Author: ok

SOURCE_MARIADB_IP=$(echo "$cpexport" | sed -e '0,/ MysqlInternal/d' | sed -n '0,/host_nobrackets/s/^.*host_nobrackets\:\s*\(.*\)$/\1/p')
export SOURCE_MARIADB_IP
+
MARIADB_IMAGE='quay.io/podified-antelope-centos9/openstack-mariadb:current-podified'
export MARIADB_IMAGE
+
RUN_OVERRIDES='{
    "apiVersion": "v1",
    "metadata": {
        "annotations": {
            "k8s.v1.cni.cncf.io/networks": "[{\"name\": \"internalapi-static\",\"namespace\": \"openstack\", \"ips\":[\"172.17.0.99/24\"]}]"
        }
    },
    "spec": {
        "nodeName": "'"$CONTROLLER_NODE"'",
        "securityContext": {
            "allowPrivilegeEscalation": false,
            "capabilities": {
                "drop": ["ALL"]
            },
            "runAsNonRoot": true,
            "seccompProfile": {
                "type": "RuntimeDefault"
            }
        }
    }
}'
----
.Procedure

. Export the shell variables for the following outputs and test the connection to the {OpenStackShort} database:
+
----

> PULL_OPENSTACK_CONFIGURATION_DATABASES="$(oc run mariadb-client -q --image "${MARIADB_IMAGE}" \

Reviewer: Should the "$" precede "PULL_OPENSTACK_CONFIGURATION_DATABASES..."?

Author: yes
